teuthology/packaging.py

```diff
@@ -970,6 +970,39 @@ def repo_url(self):
             'repo',
         )
 
+    @property
+    def build_complete(self):
+        # use the repo search results to get a ref and a sha1; the
+        # input to teuthology-suite doesn't contain both
+        try:
+            self.assert_result()
+        except VersionNotFoundError:
+            return False
+
+        search_result = self._result.json()[0]
+
+        # now look for the build complete status
+        build_url = '/'.join((
+            self.query_url.rstrip('/'),
+            'builds/ceph',
+            search_result['ref'],
+            search_result['sha1'],
+        ))
+
+        try:
+            resp = requests.get(build_url)
+            resp.raise_for_status()
+        except requests.HttpError:
+            return False
+        for build in resp.json():
+            if (
+                build['distro'] == search_result['distro'] and
+                build['distro_version'] == search_result['distro_version'] and
+                build['flavor'] == search_result['flavor']
+               ):
+                return build['status'] == 'completed'
+        return False
```

Comment:
Will there ever be `>1` builds that satisfy the above conditions? Is architecture relevant?

Reply:
I don't *think* so if I consider arch as well.  Can't hurt (I don't think; I'm pretty sure arch is available where it needs to be)